src/hatch/publish/index.py

```diff
@@ -161,13 +148,13 @@ def publish(self, artifacts: list, options: dict):
             for version in versions:
                 self.app.display_info(str(index.urls.project.child(project_name, version, '').to_iri()))
 
-        if updated_user is not None:
-            cached_user_file.set_user(repo, user)
+        if updated_user:
+            cached_user_file.set_user(repo, username)
 
-        if updated_auth is not None:
+        if updated_auth:
             import keyring
 
-            keyring.set_password(repo, user, auth)
+            keyring.set_password(repo, username, auth_token)
```

Comment:
and this logic would find its place in the `auth` module / its `Authentication` class.